main.c

```diff
@@ -74,137 +129,231 @@ int main(int argc, char **argv) {
+    uint32_t last_pres_fuel_millis = millis();
+    uint32_t last_pres_pneumatics_millis = millis();
+    uint32_t last_pres_cc_millis = millis();
+    uint32_t last_hallsense_fuel_millis = millis();
+    uint32_t last_hallsense_fill_millis = millis();
+    uint32_t last_hallsense_ox_millis = millis();
+#elif (BOARD_UNIQUE_ID == BOARD_ID_PROPULSION_VENT)
+    uint32_t last_pres_ox_millis = millis();
+    uint32_t last_vent_temp_millis = millis();
+#endif
 
-    bool blue_led_on = false;   // visual heartbeat
+    bool blue_led_on = false; // visual heartbeat
     while (1) {
         CLRWDT(); // feed the watchdog, which is set for 256ms
 
         if (seen_can_message) {
             seen_can_message = false;
             last_message_millis = millis();
         }
+        if (seen_can_command) {
+            seen_can_command = false;
+            last_command_millis = millis();
+        }
 
-        if (millis() - last_message_millis > MAX_BUS_DEAD_TIME_ms &&
-            (SAFE_STATE_ENABLED || requested_actuator_state == SAFE_STATE)) {
+#if (BOARD_UNIQUE_ID == BOARD_ID_PROPULSION_INJ)
+        if (((millis() - last_message_millis) > MAX_BUS_DEAD_TIME_ms) &&
+            (requested_actuator_state_inj == SAFE_STATE_INJ)) {
             // Only reset if safe state is enabled (aka this isn't injector valve)
             // OR this is injector valve and the currently requested state is the safe
             // state (closed)
-            
+
             // We've got too long without seeing a valid CAN message (including one of ours)
             RESET();
         }
-
+#elif (BOARD_UNIQUE_ID == BOARD_ID_PROPULSION_VENT)
+        if ((millis() - last_message_millis) > MAX_BUS_DEAD_TIME_ms) {
+            RESET();
+        }
+#endif
         if (millis() - last_status_millis > STATUS_TIME_DIFF_ms) {
             last_status_millis = millis();
 
             bool status_ok = true;
-            status_ok &= check_bus_current_error();
+            status_ok &= check_5v_current_error(current_sense_5v);
+            status_ok &= check_12v_current_error(current_sense_12v);
             if (status_ok) {
                 send_status_ok();
             }
 
             LED_heartbeat_G();
         }
-        
+
         if (millis() - last_millis > MAX_LOOP_TIME_DIFF_ms) {
 
             // check for general board status
             bool status_ok = true;
-            status_ok &= check_battery_voltage_error();
-            status_ok &= check_bus_current_error();
-            status_ok &= check_actuator_pin_error(requested_actuator_state);
+            status_ok &= check_battery_voltage_error(batt_vol_sense);
+
+            status_ok &= check_5v_current_error(current_sense_5v); 
+            status_ok &= check_12v_current_error(current_sense_12v);
 
             // if there was an issue, a message would already have been sent out
-            if (status_ok) { send_status_ok(); }
+            if (status_ok) {
+                send_status_ok();
```

Comment:
You've got two blocks of duplicated code checking for errors and sending status ok, only have one pls :smiling_face_with_three_hearts: